AdyenComponents/Apple Pay/ApplePayComponent.swift

```diff
@@ -110,29 +110,17 @@ public class ApplePayComponent: NSObject, PresentableComponent, PaymentComponent
 
     /// Finalizes ApplePay payment after being processed by payment provider.
     /// - Parameter success: The status of the payment.
-    public func didFinalize(with success: Bool) {
-        self.success = success
-        paymentAuthorizationCompletion?(success ? .success : .failure)
-        paymentAuthorizationCompletion = nil
-    }
+    public func didFinalize(with success: Bool) { }
```

Comment:
what if ApplePayComponent  is used as an individual component, shouldn't it call `paymentAuthorizationCompletion` closure  and reset its state ?

Reply:
We are no longer going to use `paymentAuthorizationCompletion` (it is removed in this PR) on v4.
For v5 we will need to build new flow for Apple Pay to treat it as SDK Action.

In this version ApplePayComponent no longer have (need) functional `didFinalize`.

Reply:
- `to treat it as SDK Action`

have no idea what you mean :-D, its not an SDK and its not an action, its a payment component.

- `We are no longer going to use paymentAuthorizationCompletion `

also don't know what you're talking about :-)

lets discuss this, I don't feel comfortable merging this code